frappe/website/doctype/web_form/web_form.py

```diff
@@ -387,6 +387,10 @@ def accept(web_form, data):
 
 	web_form = frappe.get_doc("Web Form", web_form)
 	doctype = web_form.doc_type
+	user = frappe.session.user
+
+	if web_form.anonymous and frappe.session.user != "Guest":
+		frappe.session.user = "Guest"
```

Comment:
frappe.set_user is better :sweat_smile: 


Also payment app overrides this entire method :woozy_face: @phot0n

Reply:
yep - will need to figure out a better design for it :cry:

Reply:
we need a contextmanager API for this

```py
with session_user('Guest'):
    frappe.get_doc(...).insert()
```

cc @ankush @phot0n

Reply:
If you are logged in then frappe.set_user("Guest") will log you out

Reply:
> frappe.set_user is better 😅

If you are logged in then frappe.set_user("Guest") will log you out